### PR TITLE
Add Solution for 239 Sliding Window Maximum

### PR DESCRIPTION
**Pull request by kuangami**
@yiru941013

### CODE REVIEW

File: cpp/239_Sliding_Window_Maximum.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * [1  3  -1] -3  5  3  6  7       3
+ *  1 [3  -1  -3] 5  3  6  7       3
+ *  1  3 [-1  -3  5] 3  6  7       5
+ *  1  3  -1 [-3  5  3] 6  7       5
+ *  1  3  -1  -3 [5  3  6] 7       6
+ *  1  3  -1  -3  5 [3  6  7]      7
+ * Therefore, return the max sliding window as [3,3,5,5,6,7].
+ * 
+ * Note: 
+ * You may assume k is always valid, ie: 1 �� k �� input array's size for non-empty array.
```
Comment (yiru941013) [inline cpp/239_Sliding_Window_Maximum.cpp]: 乱码 不会写英语

File: cpp/239_Sliding_Window_Maximum.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * 
+ * Follow up:
+ * Could you solve it in linear time?
+ * 
+ * Hint:
+ * 1. How about using a data structure such as deque (double-ended queue)?
+ * 2. The queue size need not be the same as the window��s size.
```
Comment (yiru941013) [inline cpp/239_Sliding_Window_Maximum.cpp]: 乱码+1 不会写英语

File: cpp/239_Sliding_Window_Maximum.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ *  1  3  -1 [-3  5  3] 6  7       5
+ *  1  3  -1  -3 [5  3  6] 7       6
+ *  1  3  -1  -3  5 [3  6  7]      7
+ * Therefore, return the max sliding window as [3,3,5,5,6,7].
+ * 
+ * Note: 
+ * You may assume k is always valid, ie: 1 �� k �� input array's size for non-empty array.
+ * 
+ * Follow up:
+ * Could you solve it in linear time?
```
Comment (yiru941013) [inline cpp/239_Sliding_Window_Maximum.cpp:22]: 这句不懂

File: cpp/239_Sliding_Window_Maximum.cpp (hunk 1 above)
Comment (kuangami) [reply cpp/239_Sliding_Window_Maximum.cpp]: You can say "strange/funny characters" or simply reply "please fix this".

File: cpp/239_Sliding_Window_Maximum.cpp (hunk 3 above)
Comment (kuangami) [reply cpp/239_Sliding_Window_Maximum.cpp:22]: It requires the time complexity O(n), which means there is no nested loop in the code.

**Review by yiru941013 — APPROVED** [review]

**Pull request merged**